### PR TITLE
Article.js: allow updates to selected props to trigger new chapter select + Article Control/Scrollbar Overlap Fix

### PR DESCRIPTION
**Pull request by jwijay**
* Article.js: allow updates to selected props to trigger new chapter select
* using margin instead of padding for article control to prevent arrow from overlapping scrollbar and hindering scrollbar usage

### CODE REVIEW

File: src/scss/grommet-core/_objects.article.scss [hunk 1]
```diff
@@ -58,9 +58,13 @@
...
   position: fixed;
   z-index: 10;
+  // using margin instead of padding for article control
+  // to prevent arrow from overlapping scrollbar and
+  // hindering scrollbar usage
+  margin: $inuit-base-spacing-unit;
 
   .button__icon {
-    padding: $inuit-base-spacing-unit;
+    padding: 0;
```
Comment (ericsoderberghp) [inline src/scss/grommet-core/_objects.article.scss]: Does this mean that the hit target for the button is no longer 48x48?

File: src/js/components/Article.js [hunk 2]
```diff
@@ -77,6 +77,13 @@ export default class Article extends Component {
     this._onSelect(this.state.selectedIndex);
   }
 
+  componentWillReceiveProps (nextProps) {
+    // allow updates to selected props to trigger new chapter select
+    if ((typeof nextProps.selected !== 'undefined') && (nextProps.selected !== this.state.selectedIndex)) {
```
Comment (ericsoderberghp) [inline src/js/components/Article.js]: Should we account for 'null' as well?

File: src/scss/grommet-core/_objects.article.scss (hunk 1 above)
Comment (jwijay) [inline src/scss/grommet-core/_objects.article.scss]: Previously the hit target was 96x96. After this PR, the hit target will be 48x48.

Before:
screen shot 2016-05-31 at 7 25 50 am

After (48x48 hit target):
screen shot 2016-05-31 at 7 23 33 am

File: src/js/components/Article.js (hunk 2 above)
Comment (jwijay) [inline src/js/components/Article.js]: Updated the PR to account for null selected props

**Pull request merged**